### PR TITLE
utils/win32: trap invalid --icon arguments and terminate with a message

### PR DESCRIPTION
**Pull request by tallforasmurf**
An invalid file path argument to --icon is detected in one of two places. If it ends in `.ico` an attempt to open the file is in the `__init__()` method of the `IconFile` class. Here trap any IOError subclass, which would include nonexistent file, unreadable file, etc, and raise SystemExit with a simple "could not open" message.

Other file extensions are opened using win32api.LoadLibraryEx which expects to get an EXE (or a DLL?). If it fails, win32api raises its unique exception which includes the windows error string and error code. These are formatted into a message string for SystemExit.

### CODE REVIEW

File: PyInstaller/utils/win32/icon.py [hunk 1]
```diff
@@ -169,8 +180,21 @@ def splitter(s):
...
+        # .ico/.exe) then LoadLibraryEx returns a null handle and win32api
+        # raises a unique exception with a win error code and a string.
+        hsrc = win32api.LoadLibraryEx(srcpath, 0, LOAD_LIBRARY_AS_DATAFILE)
+    except win32api.error as W32E:
+        # We could continue with no icon (i.e. just return) however it seems
+        # best to terminate the build with a message.
+        raise SystemExit(
+            "Unable to load icon file {}\n    {} (Error code {})".format(
+                srcpath, W32E.strerror, W32E.winerror)
+            )
```
Comment (xoviat) [inline PyInstaller/utils/win32/icon.py]: This line is still indented too much.

**Comment by tallforasmurf** [comment]
Based on advice in [this blog post](https://www.burntfen.com/2015-10-30/how-to-amend-a-commit-on-a-github-pull-request) I did the following:
* corrected spacing and saved file

    git add -A
    git commit --amend
    git push -f origin HEAD

Clicking the "View changes" button above gives the scary message "We went looking everywhere, but couldn’t find those commits." However clicking the link for 7163b17 shows the correct stuff.

**Comment by xoviat** [comment]
@tallforasmurf Honestly, I wouldn't worry about it, but your workflow is still not correct. The HEAD branch should NOT be named "develop." You should have checked out a different branch with a descriptive name for this feature.

However, IMO it's not a big deal and not worth it to try again (and possibly lose work).

**Comment by htgoebel** [comment]
Thanks :-)

**Pull request merged**